binding.cc

```diff
@@ -1047,21 +1047,25 @@ namespace zmq {
     private:
       class BufferReference {
         public:
-          inline BufferReference(Local<Object> buf) {
-            async = new uv_async_t;
-            uv_async_init(uv_default_loop(), async, Destroy);
-            async->data = this;
-            persistent.Reset(buf);
+          inline BufferReference(Local<Object> buf)
+            : persistent_(buf), async_(new uv_async_t) {
+            if (uv_async_init(uv_default_loop(), async_, Destroy) < 0) {
+              delete async_;
+              delete this;
+              Nan::ThrowError("Async initialization failed");
```

Comment:
None of the `ThrowXxx` actually break control flow, so the succeeding line assigning `this` to `async_->data` will still execute. Is this intentional?

Reply:
Definitely not, that is an oversight!